Supporting storing and loading pipeline states as part of the multi-timeline

shankari opened the pull request:
Since the multi-timeline can contain analysis results, if we want to correctly transfer the entire pipeline state to a live system (e.g. from proprietary -> open) we need to store and re-load the pipeline state as well.

AmplabJenkins commented:
Build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/e-mission-server-prb/994/
Test FAILed.

shankari commented:
Loading only the raw data and re-running the pipeline multiple times works.

```
$ ./e-mission-py.bash bin/purge_database.py
storage not configured, falling back to sample, default configuration
Connecting to database URL localhost
debug not configured, falling back to sample, default configuration
WARNING:root:google maps key not configured, falling back to nominatim
collections = ['Stage_Profiles', 'Stage_pipeline_state', 'Stage_uuids', 'Stage_analysis_timeseries', 'Stage_timeseries', 'Stage_usercache']

(emission) C02KT61MFFT0:e-mission-server shankari$ ./e-mission-py.bash bin/debug/load_timeline_for_day_and_user.py emission/tests/data/real_examples/shankari_single_positional_indexer.dec-12 test-single-pipeline
storage not configured, falling back to sample, default configuration
Connecting to database URL localhost
emission/tests/data/real_examples/shankari_single_positional_indexer.dec-12
Loading file emission/tests/data/real_examples/shankari_single_positional_indexer.dec-12
After registration, test-single-pipeline -> 9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5

(emission) C02KT61MFFT0:e-mission-server shankari$ ./e-mission-py.bash bin/debug/intake_single_user.py -e test-single-pipeline
storage not configured, falling back to sample, default configuration
Connecting to database URL localhost
debug not configured, falling back to sample, default configuration
WARNING:root:google maps key not configured, falling back to nominatim
2018-01-14T07:20:19.237987-08:00**********UUID 9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5: moving to long term**********
2018-01-14T07:20:24.982740-08:00**********UUID 9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5: storing views to cache**********

(emission) C02KT61MFFT0:e-mission-server shankari$ ./e-mission-py.bash bin/debug/intake_single_user.py -e test-single-pipeline
storage not configured, falling back to sample, default configuration
Connecting to database URL localhost
debug not configured, falling back to sample, default configuration
WARNING:root:google maps key not configured, falling back to nominatim
2018-01-14T07:20:29.418093-08:00**********UUID 9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5: moving to long term**********
2018-01-14T07:20:29.495982-08:00**********UUID 9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5: storing views to cache**********

(emission) C02KT61MFFT0:e-mission-server shankari$ ./e-mission-py.bash bin/debug/intake_single_user.py -e test-single-pipeline
storage not configured, falling back to sample, default configuration
Connecting to database URL localhost
debug not configured, falling back to sample, default configuration
WARNING:root:google maps key not configured, falling back to nominatim
2018-01-14T07:20:32.701961-08:00**********UUID 9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5: moving to long term**********
2018-01-14T07:20:32.767774-08:00**********UUID 9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5: storing views to cache**********
(emission) C02KT61MFFT0:e-mission-server shankari$
```

AmplabJenkins commented:
Build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/e-mission-server-prb/995/
Test FAILed.

shankari commented:
$ ./e-mission-py.bash bin/debug/extract_timeline_for_day_range_and_user.py -e test-single-pipeline -- 2010-01-01 2020-01-01 /tmp/test_dump
storage not configured, falling back to sample, default configuration
Connecting to database URL localhost
INFO:root:==================================================
INFO:root:Extracting timeline for user 9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5 day 2010-01-01 -> 2020-01-01 and saving to file /tmp/test_dump
DEBUG:root:start_day_ts = 1262304000 (2010-01-01T00:00:00+00:00), end_day_ts = 1577836800 (2020-01-01T00:00:00+00:00)
...
INFO:root:Found 1449 loc entries, 27 trip-like entries, 19 place-like entries = 1495 total entries
INFO:root:timeline has unique keys = {'stats/server_api_error', 'stats/client_time', 'stats/client_nav_event', 'stats/pipeline_time', 'background/location', 'background/motion_activity', 'analysis/cleaned_trip', 'analysis/cleaned_stop', 'stats/server_api_time', 'background/filtered_location', 'segmentation/raw_stop', 'analysis/cleaned_section', 'statemachine/transition', 'analysis/cleaned_place', 'background/battery', 'segmentation/raw_section', 'segmentation/raw_place', 'segmentation/raw_trip', 'analysis/recreated_location'}
INFO:root:Found 6 pipeline states [6, 1, 2, 3, 11, 9]

$ ./e-mission-py.bash bin/debug/load_multi_timeline_for_range.py /tmp/test_dump_
storage not configured, falling back to sample, default configuration
Connecting to database URL localhost
INFO:root:Loading file or prefix /tmp/test_dump_
INFO:root:Found 2 matching files for prefix /tmp/test_dump_
INFO:root:files are ['/tmp/test_dump_9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5.gz', '/tmp/test_dump_pipelinestate_9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5.gz'] ... ['/tmp/test_dump_9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5.gz']
INFO:root:==================================================
INFO:root:Loading data from file /tmp/test_dump_9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5.gz
INFO:root:For 1 users, loaded 1272 raw entries, 223 processed entries and 6 pipeline states
INFO:root:all entries in the timeline contain analysis results, no need to run the intake pipeline

$ ./e-mission-py.bash bin/debug/intake_single_user.py -u 9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5
storage not configured, falling back to sample, default configuration
Connecting to database URL localhost
debug not configured, falling back to sample, default configuration
WARNING:root:google maps key not configured, falling back to nominatim
2018-01-14T07:33:26.690506-08:00**********UUID 9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5: moving to long term**********
2018-01-14T07:33:26.869440-08:00**********UUID 9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5: filter accuracy if needed**********
2018-01-14T07:33:26.876891-08:00**********UUID 9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5: segmenting into trips**********
2018-01-14T07:33:26.888589-08:00**********UUID 9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5: segmenting into sections**********
2018-01-14T07:33:26.898514-08:00**********UUID 9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5: smoothing sections**********
2018-01-14T07:33:26.907314-08:00**********UUID 9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5: cleaning and resampling timeline**********
2018-01-14T07:33:26.927160-08:00**********UUID 9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5: checking active mode trips to autocheck habits**********
2018-01-14T07:33:26.935356-08:00**********UUID 9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5: storing views to cache**********
Found error 'NoneType' object has no attribute 'get_id' while processing trip Entry({'_id': ObjectId('5a5b7536f6858f9b56edb328'), 'user_id': UUID('9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5'), 'metadata': {'key': 'analysis/cleaned_trip', 'platform': 'server', 'write_ts': 1515943222.629767, 'time_zone': 'America/Los_Angeles', 'write_local_dt': {'year': 2018, 'month': 1, 'day': 14, 'hour': 7, 'minute': 20, 'second': 22, 'weekday': 6, 'timezone': 'America/Los_Angeles'}, 'write_fmt_time': '2018-01-14T07:20:22.629767-08:00'}, 'data': {'source': 'DwellSegmentationTimeFilter', 'end_ts': 1481563775.615, 'end_local_dt': {'year': 2016, 'month': 12, 'day': 12, 'hour': 9, 'minute': 29, 'second': 35, 'weekday': 0, 'timezone': 'America/Los_Angeles'}, 'end_fmt_time': '2016-12-12T09:29:35.615000-08:00', 'end_loc': {'type': 'Point', 'coordinates': [-122.0850374, 37.4029096]}, 'raw_trip': ObjectId('5a5b7535f6858f9b56edb2ff'), 'start_ts': 1481563149.784, 'start_local_dt': {'year': 2016, 'month': 12, 'day': 12, 'hour': 9, 'minute': 19, 'second': 9, 'weekday': 0, 'timezone': 'America/Los_Angeles'}, 'start_fmt_time': '2016-12-12T09:19:09.784000-08:00', 'start_loc': {'type': 'Point', 'coordinates': [-122.0858305, 37.3918544]}, 'duration': 625.8310000896454, 'distance': 1972.6993136317851, 'start_place': ObjectId('5a5b7537f6858f9b56edb3e9'), 'end_place': ObjectId('5a5b7538f6858f9b56edb3ea')}})
Traceback (most recent call last):
  File "/Users/shankari/e-mission/e-mission-server/emission/analysis/plotting/geojson/geojson_feature_converter.py", line 269, in get_geojson_for_timeline
    trip_geojson = trip_to_geojson(trip, tl)
  File "/Users/shankari/e-mission/e-mission-server/emission/analysis/plotting/geojson/geojson_feature_converter.py", line 214, in trip_to_geojson
    start_place_geojson = place_to_geojson(curr_start_place)
  File "/Users/shankari/e-mission/e-mission-server/emission/analysis/plotting/geojson/geojson_feature_converter.py", line 74, in place_to_geojson
    ret_feature.id = str(place.get_id())
AttributeError: 'NoneType' object has no attribute 'get_id'
Storing views to cache failed for user 9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5
Traceback (most recent call last):
  File "/Users/shankari/e-mission/e-mission-server/emission/net/usercache/builtin_usercache_handler.py", line 109, in storeViewsToCache
    self.storeTimelineToCache(time_query)
  File "/Users/shankari/e-mission/e-mission-server/emission/net/usercache/builtin_usercache_handler.py", line 180, in storeTimelineToCache
    trip_gj_list = self.get_trip_list_for_seven_days(start_ts)
  File "/Users/shankari/e-mission/e-mission-server/emission/net/usercache/builtin_usercache_handler.py", line 269, in get_trip_list_for_seven_days
    trip_gj_list = gfc.get_geojson_for_ts(self.user_id, seventy_two_hours_ago_ts, start_ts)
  File "/Users/shankari/e-mission/e-mission-server/emission/analysis/plotting/geojson/geojson_feature_converter.py", line 253, in get_geojson_for_ts
    return get_geojson_for_timeline(user_id, tl)
  File "/Users/shankari/e-mission/e-mission-server/emission/analysis/plotting/geojson/geojson_feature_converter.py", line 274, in get_geojson_for_timeline
    raise e
  File "/Users/shankari/e-mission/e-mission-server/emission/analysis/plotting/geojson/geojson_feature_converter.py", line 269, in get_geojson_for_timeline
    trip_geojson = trip_to_geojson(trip, tl)
  File "/Users/shankari/e-mission/e-mission-server/emission/analysis/plotting/geojson/geojson_feature_converter.py", line 214, in trip_to_geojson
    start_place_geojson = place_to_geojson(curr_start_place)
  File "/Users/shankari/e-mission/e-mission-server/emission/analysis/plotting/geojson/geojson_feature_converter.py", line 74, in place_to_geojson
    ret_feature.id = str(place.get_id())
AttributeError: 'NoneType' object has no attribute 'get_id'

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/e-mission-server-prb/996/
Test FAILed.

shankari commented:
Error is while converting the start place

```
Traceback (most recent call last):
  File "/Users/shankari/e-mission/e-mission-server/emission/analysis/plotting/geojson/geojson_feature_converter.py", line 269, in get_geojson_for_timeline
    trip_geojson = trip_to_geojson(trip, tl)
  File "/Users/shankari/e-mission/e-mission-server/emission/analysis/plotting/geojson/geojson_feature_converter.py", line 214, in trip_to_geojson
    start_place_geojson = place_to_geojson(curr_start_place)
  File "/Users/shankari/e-mission/e-mission-server/emission/analysis/plotting/geojson/geojson_feature_converter.py", line 74, in place_to_geojson
    ret_feature.id = str(place.get_id())
AttributeError: 'NoneType' object has no attribute 'get_id'
```

of 

2018-01-14 07:33:26,948:ERROR:140735906452416:Found error 'NoneType' object has no attribute 'get_id' while processing trip Entry({'_id': ObjectId('5a5b7536f6858f9b56edb328'), 'user_id': UUID('9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5'), 'metadata': {'key': 'analysis/cleaned_trip'}, 'data': {'source': 'DwellSegmentationTimeFilter', 'end_ts': 1481563775.615, 'end_fmt_time': '2016-12-12T09:29:35.615000-08:00', 'end_loc': {'type': 'Point', 'coordinates': [-122.0850374, 37.4029096]}, 'raw_trip': ObjectId('5a5b7535f6858f9b56edb2ff'), 'start_ts': 1481563149.784, 'start_fmt_time': '2016-12-12T09:19:09.784000-08:00', 'start_loc': {'type': 'Point', 'coordinates': [-122.0858305, 37.3918544]}, 'duration': 625.8310000896454, 'distance': 1972.6993136317851, 'start_place': ObjectId('5a5b7537f6858f9b56edb3e9'), 'end_place': ObjectId('5a5b7538f6858f9b56edb3ea')}})

Let's see what happened to the place with id `5a5b7537f6858f9b56edb3e9`.

It is not in the map or in the database.

```
2018-01-14 07:33:26,947:DEBUG:140735906452416:for trip = 5a5b7536f6858f9b56edb328, adding start_place 5a5b7537f6858f9b56edb3e9 and end_place 5a5b7538f6858f9b56edb3ea
2018-01-14 07:33:26,947:DEBUG:140735906452416:place id 5a5b7537f6858f9b56edb3e9 is not in the map, searching in database
2018-01-14 07:33:26,947:DEBUG:140735906452416:retrieved object None and added to id_map
```

But on a previous run, it was in the database.

```
2018-01-14 07:20:32,778:DEBUG:140735906452416:for trip = 5a5b7536f6858f9b56edb328, adding start_place 5a5b7537f6858f9b56edb3e9 and end_place 5a5b7538f6858f9b56edb3ea
2018-01-14 07:20:32,778:DEBUG:140735906452416:place id 5a5b7537f6858f9b56edb3e9 is not in the map, searching in database
2018-01-14 07:20:32,780:DEBUG:140735906452416:retrieved object Entry({'_id': ObjectId('5a5b7537f6858f9b56edb3e9'), 'user_id': UUID('9cc1abbc-ec4d-4221-9cf6-eea46fee6aa5'), 'metadata': {'key': 'analysis/cleaned_place', 'platform': 'server', 'write_ts': 1515943223.9811578, 'time_zone': 'America/Los_Angeles', 'write_local_dt': {'year': 2018, 'month': 1, 'day': 14, 'hour': 7, 'minute': 20, 'second': 23, 'weekday': 6, 'timezone': 'America/Los_Angeles'}, 'write_fmt_time': '2018-01-14T07:20:23.981158-08:00'}, 'data': {'source': 'DwellSegmentationTimeFilter', 'location': {'type': 'Point', 'coordinates': [-122.0858305, 37.3918544]}, 'raw_places': [ObjectId('5a5b7535f6858f9b56edb2fe'), ObjectId('5a5b7535f6858f9b56edb2fe')], 'display_name': 'Mercy Street, Mountain View', 'starting_trip': ObjectId('5a5b7536f6858f9b56edb328'), 'exit_ts': 1481563149.784, 'exit_fmt_time': '2016-12-12T09:19:09.784000-08:00', 'exit_local_dt': {'year': 2016, 'month': 12, 'day': 12, 'hour': 9, 'minute': 19, 'second': 9, 'weekday': 0, 'timezone': 'America/Los_Angeles'}}}) and added to id_map
```

Aha! It doesn't have a `data.enter_ts`, probably because it was the first place
that starts the timeline and so it doesn't match the criterion of 

```
    place_time_query = estt.TimeQuery("data.enter_ts", start_day_ts, end_day_ts)
    place_entry_list = list(ts.find_entries(key_list=None, time_query=place_time_query))
```

I am not using hardcoded queries, so it is unclear how I can include
`{'data.enter_ts': {'$eq': None}}`. Maybe use the extra_query_list?

shankari commented:
`extra_query_list` dumps fine but apparently generates duplicate key errors.

shankari commented:
Extract values

```
$ ./e-mission-py.bash bin/debug/extract_timeline_for_day_range_and_user.py -e test-single-pipeline -- 2010-01-01 2020-01-01 /tmp/test_dump
storage not configured, falling back to sample, default configuration
Connecting to database URL localhost
INFO:root:==================================================
INFO:root:Extracting timeline for user 23d2e860-7fe1-4510-904d-bdb34dd70f13 day 2010-01-01 -> 2020-01-01 and saving to file /tmp/test_dump
DEBUG:root:start_day_ts = 1262304000 (2010-01-01T00:00:00+00:00), end_day_ts = 1577836800 (2020-01-01T00:00:00+00:00)
...
DEBUG:root:curr_query = {'user_id': UUID('23d2e860-7fe1-4510-904d-bdb34dd70f13'), '$and': [{'data.enter_ts': {'$exists': False}}, {'data.exit_ts': {'$exists': True}}]}, sort_key = metadata.write_ts
DEBUG:root:orig_ts_db_keys = None, analysis_ts_db_keys = None
DEBUG:root:finished querying values for None
DEBUG:root:finished querying values for None
```

Two first places, one raw and one cleaned.

```
INFO:root:First place entry list = [{'_id': ObjectId('5a5c3eb8f6858fad8d59b156'), 'user_id': UUID('23d2e860-7fe1-4510-904d-bdb34dd70f13'), 'metadata': {'key': 'segmentation/raw_place', 'platform': 'server', 'write_ts': 1515994808.52354, 'time_zone': 'America/Los_Angeles', 'write_local_dt': {'year': 2018, 'month': 1, 'day': 14, 'hour': 21, 'minute': 40, 'second': 8, 'weekday': 6, 'timezone': 'America/Los_Angeles'}, 'write_fmt_time': '2018-01-14T21:40:08.523540-08:00'}, 'data': {'source': 'DwellSegmentationTimeFilter', 'exit_ts': 1481563149.784, 'exit_local_dt': {'hour': 9, 'month': 12, 'second': 9, 'weekday': 0, 'year': 2016, 'timezone': 'America/Los_Angeles', 'day': 12, 'minute': 19}, 'exit_fmt_time': '2016-12-12T09:19:09.784000-08:00', 'starting_trip': ObjectId('5a5c3eb8f6858fad8d59b157'), 'location': {'type': 'Point', 'coordinates': [-122.0858305, 37.3918544]}}},
{'_id': ObjectId('5a5c3ebaf6858fad8d59b241'), 'user_id': UUID('23d2e860-7fe1-4510-904d-bdb34dd70f13'), 'metadata': {'key': 'analysis/cleaned_place', 'platform': 'server', 'write_ts': 1515994810.5758588, 'time_zone': 'America/Los_Angeles', 'write_local_dt': {'year': 2018, 'month': 1, 'day': 14, 'hour': 21, 'minute': 40, 'second': 10, 'weekday': 6, 'timezone': 'America/Los_Angeles'}, 'write_fmt_time': '2018-01-14T21:40:10.575859-08:00'}, 'data': {'source': 'DwellSegmentationTimeFilter', 'location': {'type': 'Point', 'coordinates': [-122.0858305, 37.3918544]}, 'raw_places': [ObjectId('5a5c3eb8f6858fad8d59b156'), ObjectId('5a5c3eb8f6858fad8d59b156')], 'display_name': 'Mercy Street, Mountain View', 'starting_trip': ObjectId('5a5c3eb9f6858fad8d59b180'), 'exit_ts': 1481563149.784, 'exit_fmt_time': '2016-12-12T09:19:09.784000-08:00', 'exit_local_dt': {'year': 2016, 'month': 12, 'day': 12, 'hour': 9, 'minute': 19, 'second': 9, 'weekday': 0, 'timezone': 'America/Los_Angeles'}}}]
```

Extract complete

```
INFO:root:Found 1433 loc entries, 27 trip-like entries, 19 place-like entries = 1481 total entries
INFO:root:timeline has unique keys = {'stats/server_api_time', 'analysis/cleaned_place', 'analysis/cleaned_stop', 'statemachine/transition', 'background/filtered_location', 'segmentation/raw_place', 'segmentation/raw_stop', 'analysis/cleaned_trip', 'stats/client_time', 'stats/pipeline_time', 'stats/server_api_error', 'background/motion_activity', 'segmentation/raw_trip', 'segmentation/raw_section', 'analysis/cleaned_section', 'background/battery', 'background/location', 'analysis/recreated_location', 'stats/client_nav_event'}
INFO:root:Found 6 pipeline states [6, 1, 2, 3, 11, 9]
```

Next, purge everything

```
$ ./e-mission-py.bash bin/purge_database.py
storage not configured, falling back to sample, default configuration
Connecting to database URL localhost
debug not configured, falling back to sample, default configuration
WARNING:root:google maps key not configured, falling back to nominatim
collections = ['Stage_Profiles', 'Stage_pipeline_state', 'Stage_uuids', 'Stage_analysis_timeseries', 'Stage_timeseries', 'Stage_usercache']
```

Next, reload

```
$ ./e-mission-py.bash bin/debug/load_multi_timeline_for_range.py /tmp/test_dump_
storage not configured, falling back to sample, default configuration
Connecting to database URL localhost
INFO:root:Loading file or prefix /tmp/test_dump_
INFO:root:Found 2 matching files for prefix /tmp/test_dump_
INFO:root:files are ['/tmp/test_dump_23d2e860-7fe1-4510-904d-bdb34dd70f13.gz', '/tmp/test_dump_pipelinestate_23d2e860-7fe1-4510-904d-bdb34dd70f13.gz'] ... ['/tmp/test_dump_23d2e860-7fe1-4510-904d-bdb34dd70f13.gz']
INFO:root:==================================================
INFO:root:Loading data from file /tmp/test_dump_23d2e860-7fe1-4510-904d-bdb34dd70f13.gz
INFO:root:Analyzing timeline...
INFO:root:timeline has 1481 entries
INFO:root:timeline has data from 1 users
INFO:root:timeline has the following unique keys {'background/filtered_location', 'background/location', 'background/motion_activity', 'analysis/cleaned_section', 'stats/client_nav_event', 'stats/client_time', 'background/battery', 'segmentation/raw_trip', 'analysis/cleaned_place', 'stats/server_api_time', 'statemachine/transition', 'analysis/cleaned_trip', 'stats/pipeline_time', 'segmentation/raw_section', 'segmentation/raw_place', 'stats/server_api_error', 'analysis/cleaned_stop', 'segmentation/raw_stop', 'analysis/recreated_location'}
INFO:root:timeline for user 23d2e860-7fe1-4510-904d-bdb34dd70f13 contains analysis results
Loading pipeline state for 23d2e860-7fe1-4510-904d-bdb34dd70f13 from /tmp/test_dump__pipelinestate_23d2e860-7fe1-4510-904d-bdb34dd70f13.gz
INFO:root:Creating user entries for 1 users
INFO:root:pattern = user-%01d
INFO:root:For 1 users, loaded 1256 raw entries, 225 processed entries and 6 pipeline states
INFO:root:all entries in the timeline contain analysis results, no need to run the intake pipeline
```

Then, re-run pipeline, no errors.

```
$ ./e-mission-py.bash bin/debug/intake_single_user.py -u 23d2e860-7fe1-4510-904d-bdb34dd70f13
storage not configured, falling back to sample, default configuration
Connecting to database URL localhost
debug not configured, falling back to sample, default configuration
WARNING:root:google maps key not configured, falling back to nominatim
2018-01-14T22:02:30.332175-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: moving to long term**********
2018-01-14T22:02:30.506501-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: filter accuracy if needed**********
2018-01-14T22:02:30.512928-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: segmenting into trips**********
2018-01-14T22:02:30.526333-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: segmenting into sections**********
2018-01-14T22:02:30.534371-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: smoothing sections**********
2018-01-14T22:02:30.542478-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: cleaning and resampling timeline**********
2018-01-14T22:02:30.555167-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: checking active mode trips to autocheck habits**********
2018-01-14T22:02:30.567254-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: storing views to cache**********
```

Then, re-run pipeline again, still no errors.

```
$ ./e-mission-py.bash bin/debug/intake_single_user.py -u 23d2e860-7fe1-4510-904d-bdb34dd70f13
storage not configured, falling back to sample, default configuration
Connecting to database URL localhost
debug not configured, falling back to sample, default configuration
WARNING:root:google maps key not configured, falling back to nominatim
2018-01-14T22:02:33.875691-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: moving to long term**********
2018-01-14T22:02:33.887238-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: filter accuracy if needed**********
2018-01-14T22:02:33.893846-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: segmenting into trips**********
2018-01-14T22:02:33.903650-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: segmenting into sections**********
2018-01-14T22:02:33.911195-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: smoothing sections**********
2018-01-14T22:02:33.919707-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: cleaning and resampling timeline**********
2018-01-14T22:02:33.928721-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: checking active mode trips to autocheck habits**********
2018-01-14T22:02:33.931759-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: storing views to cache**********
```

And third time too.

I now declare *victory*

```
(emission) C02KT61MFFT0:e-mission-server shankari$ ./e-mission-py.bash bin/debug/intake_single_user.py -u 23d2e860-7fe1-4510-904d-bdb34dd70f13
storage not configured, falling back to sample, default configuration
Connecting to database URL localhost
debug not configured, falling back to sample, default configuration
WARNING:root:google maps key not configured, falling back to nominatim
2018-01-14T22:02:37.363801-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: moving to long term**********
2018-01-14T22:02:37.375991-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: filter accuracy if needed**********
2018-01-14T22:02:37.385132-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: segmenting into trips**********
2018-01-14T22:02:37.397237-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: segmenting into sections**********
2018-01-14T22:02:37.408634-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: smoothing sections**********
2018-01-14T22:02:37.418842-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: cleaning and resampling timeline**********
2018-01-14T22:02:37.429425-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: checking active mode trips to autocheck habits**********
2018-01-14T22:02:37.432917-08:00**********UUID 23d2e860-7fe1-4510-904d-bdb34dd70f13: storing views to cache**********
```

works now!

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/e-mission-server-prb/997/
Test PASSed.
